widget/toolbar.go

```diff
@@ -17,27 +17,51 @@ type ToolbarItem interface {
 type ToolbarAction struct {
 	Icon        fyne.Resource
 	OnActivated func() `json:"-"`
+	button      *Button
 }
 
 // ToolbarObject gets a button to render this ToolbarAction
 func (t *ToolbarAction) ToolbarObject() fyne.CanvasObject {
-	button := NewButtonWithIcon("", t.Icon, t.OnActivated)
-	button.Importance = LowImportance
-
-	return button
+	t.button.OnTapped = t.OnActivated
+	return t.button
 }
 
 // SetIcon updates the icon on a ToolbarItem
 //
 // Since: 2.2
 func (t *ToolbarAction) SetIcon(icon fyne.Resource) {
 	t.Icon = icon
-	t.ToolbarObject().Refresh()
+	t.button.SetIcon(t.Icon)
+	t.button.Refresh()
+}
+
+// Enable this ToolbarAction, updating any style or features appropriately.
+//
+// Since: 2.5
+func (t *ToolbarAction) Enable() {
+	t.button.Enable()
+}
+
+// Disable this ToolbarAction so that it cannot be interacted with, updating any style appropriately.
+//
+// Since: 2.5
+func (t *ToolbarAction) Disable() {
+	t.button.Disable()
+}
+
+// Disabled returns true if this ToolbarAction is currently disabled or false if it can currently be interacted with.
+//
+// Since: 2.5
+func (t *ToolbarAction) Disabled() bool {
+	return t.button.Disabled()
 }
 
 // NewToolbarAction returns a new push button style ToolbarItem
 func NewToolbarAction(icon fyne.Resource, onActivated func()) *ToolbarAction {
-	return &ToolbarAction{icon, onActivated}
+	button := NewButtonWithIcon("", icon, onActivated)
```

Comment:
The Fyne APIs are designed to support initialization using the constructor or using the struct directly (mostly for extending widgets), so you can't assume the button will have been initialized the first time that SetIcon/Enable/etc is called - so perhaps add a private helper function `ensureButton` that will construct the button if it is nil.

Reply:
Understood.

Another option could also be to not save a pointer to the button, but the button itself, right? If the ToolbarAction is created via struct initializer directly, this way the button would automatically be in a valid state. Of course, if we want to keep the button embedded into the ToolbarAction type, then the state of the ToolbarAction would need to be synchronized in the ToolbarObject() call (since if the Icon is initialized via the struct initialization, currently it is not applied to the Button)